specs/logging_implementation.adoc

```diff
@@ -0,0 +1,174 @@
+= Specification for logging implementation
+
+Logs and alerts need to be treated as messages emitted by a component.
+
+== Problem description
+
+* Logs and alerts need to be treated as messages emitted by a component.
+* A schema needs to be defined for emitted messages. This could be a json schema.
+  The schema needs to include metadata that allows the exact point of origin of
+  the message to be known.
+* Message priorities, types etc. need to be defined at the time of emission.
+* The integration/bridge applications must parse and restructure messages emitted
+  by the storage system to fit in the tendrl message schema before emitting it.
+* For messages that serve as notifications to the administrator (or alerts),
+  exact error message must be hardcoded by the developers. No backtraces or
+  system generated error messages should be sent as notifications to the
+  administrator.
+* Notifications sent to the administrator should provide sufficient context to
+  enable debugging of the problem via investigation into the log files.
+* Notifications and alerts need to be delivered to the node agent alert sockets.
+* Commons needs to implement a logging library that hides all the implementation
+  details and configuration options from any of the components using the library.
+  Components should be able to blindly emit messages to the common logging module
+  and it should be able to deliver them to either syslog, log files, alert
+  sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Use Cases
+
+Logs and alerts need to be treated as messages.
+
+== Proposed change
+* Create a new class called "Message" in common with attributes name, type,
+  message, priority(empty for log), alert_id ,log_level(empty for alert),
```

Comment:
1.ya i have toadd timestamp also,
2.I have declared class called Message for holding the message and other details. And alert and log can be handled from that object.
3.
    * yes log_level have 1-1 mapping to syslog's : log level also came from object, with the help of log level i am calling syslog function.
@brainfunked your comments pleases